specs/mgt-login.md

```diff
@@ -0,0 +1,124 @@
+| The user clicks the sign out button | <ul><li> **If >1 accounts are signed in**: The user is redirected to the authentication flow and is prompted to select the account they intend to sign out of. If the current active account is selected, the next inactive account is assigned to be the active account. The user is redirected to the application with all components refreshed with data for the newly active account. If the user selects an inactive account, the active account remains the same and the user is redirected back to the application. </li><li> **If only 1 account is signed in**: The user is taken through the sign out flow and then redirected to the home screen of the application. The login component provides a sign in button and all other components are re-rendered without user data </li></ul>|
+
+## Supported Functionality
+
+| Feature | Priority | Notes |
+| ------- | -------- | ----- |
+| **v1** | | | 
+| Render the appropriate version of the login component depending on which authentication provider is used | P0 | |
+| Render a sign in button when no users are signed in | P0 | |
+| Initiate the Identity platform authentication process when the sign in button or add another account is clicked | P0 | |
+| Assign an account as active on successful sign in | P0 | |
+| Display profile details for the active account in the signed in state | P0 | |
+| Render the flyout when the component is clicked in the signed in state | P0 | |
+| Display profile details for all signed in accounts in the flyout | P0 | |
+| Provide a sign out button which initiates the sign out flow | P0 | |
+| Properly reassign the active account and notify all other components when the user initates a change (selection of an inactive account or sign out of the active account) | P0 | |
+| Provide templates for the developer to configure content rendered in the button and the flyout | P1 | |
+| **v2** | | |
+| Provide a view attribute/property to easily change the rendering of component in the signed in state | P2 | |
+| Provide an attribute/property to configure additional actions/buttons inside of the flyout | P2 | |
+
+
+## Proposed Solution
+
+### Example 1: Login component is used with MSAL 1.0 Provider
+
+```
+<mgt-msal-provider client-id="123"></mgt-msal-provider>
+<mgt-login></mgt-login>
+```
+
+Single account version of Login component is rendered.
+
+### Example 2: Login component is used with MSAL 2.0 Provider
+
+```
+<mgt-msal-2-provider client-id="123"></mgt--msal-2-provider>
+<mgt-login></mgt-login>
+```
+
+Multiple account version of the Login component is rendered.
+
+### Example 3: Login component is used with no authentication provider
+
+```
+<mgt-login id="myLoginControl"></mgt-login>
+
+<script>
+let loginControl = document.getElementById('myLoginControl');
+loginControl.userDetails = {
+        displayName: 'Elise Yang',
+        mail: 'elise@contoso.com',
+        personImage: 'url'
+    }
+</script>
+```
```

Comment:
We decided that multi account will be disabled here, right?

Reply:
Agree, multi account should only be available via a provider

Reply:
Added note that this scenario will render the single account view in the flyout.